OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnPrerequisite.cs

```diff
@@ -62,6 +61,11 @@ void INotifyRemovedFromWorld.RemovedFromWorld(Actor self)
 				globalManager.Unregister(self, this, info.Prerequisites);
 		}
 
+		void INotifyOwnerChanged.OnOwnerChanged(Actor self, Player oldOwner, Player newOwner)
+		{
+			globalManager = newOwner.PlayerActor.Trait<GrantConditionOnPrerequisiteManager>();
```

Comment:
Ah, alright then. I had only added loggin inside `PrerequisitesUpdated` and saw it being canceled there (somewhen). Thanks for having a look anyway. :)